src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,52 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final int INCREASES_INDEX = 2;
+    private T[] arrayOfObjects;
+    private int size;
+
+    public MateStack() {
+        size = 0;
+        arrayOfObjects = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     public void push(T value) {
+        increaseCapacity();
```

Comment:
is this method always increases capacity? maybe we should separate check from grow?